Sources/RemindersLibrary/Reminders.swift

```diff
@@ -16,6 +16,60 @@ private extension EKReminder {
     }
 }
 
+struct ReminderData: Encodable {
+    private enum CodingKeys: String, CodingKey {
+        case id
+        case calendarTitle
+        case title
+        case creationDate
+        case lastModifiedDate
+        case startDate
+        case dueDate
+        case notes
+        case priority
+        case isCompleted
+        case completionDate
+        case alarms
+        case recurrenceRules
+    }
+
+    let id: String
+    let calendarTitle: String?
+    let title: String?
+    let creationDate: Date?
+    let lastModifiedDate: Date?
+    let startDate: Date?
+    let dueDate: Date?
+    let notes: String?
+    let priority: Int
+    let isCompleted: Bool
+    let completionDate: Date?
+    let alarms: [EKAlarm]?
+    let recurrenceRules: [EKRecurrenceRule]?
+
+    public func encode(to encoder: Encoder) throws {
+        var container = encoder.container(keyedBy: CodingKeys.self)
+
+        try container.encode(id, forKey: .id)
+        try container.encode(calendarTitle, forKey: .calendarTitle)
+        try container.encode(title, forKey: .title)
+        try container.encode(creationDate.map { $0.formatted(Date.ISO8601FormatStyle()) }, forKey: .creationDate)
+        try container.encode(lastModifiedDate.map { $0.formatted(Date.ISO8601FormatStyle()) }, forKey: .lastModifiedDate)
+        try container.encode(startDate.map { $0.formatted(Date.ISO8601FormatStyle()) }, forKey: .startDate)
+        try container.encode(dueDate.map { $0.formatted(Date.ISO8601FormatStyle()) }, forKey: .dueDate)
+        try container.encode(notes, forKey: .notes)
+        try container.encode(priority, forKey: .priority)
+        try container.encode(isCompleted, forKey: .isCompleted)
+        try container.encode(completionDate.map { $0.formatted(Date.ISO8601FormatStyle()) }, forKey: .completionDate)
+        // TODO: error: referencing instance method 'encode(_:forKey:)' on 'Array' requires that 'EKAlarm' conform to 'Encodable'
+        //   Ref: https://developer.apple.com/documentation/eventkit/ekalarm
+        // try container.encode(alarms, forKey: .alarms)
+        // TODO: error: referencing instance method 'encode(_:forKey:)' on 'Array' requires that 'EKRecurrenceRule' conform to 'Encodable'
+        //   Ref: https://developer.apple.com/documentation/eventkit/ekrecurrencerule
+        // try container.encode(recurrenceRules, forKey: .recurrenceRules)
```

Comment:
TODO

Reply:
probably fine to omit these for now, getting these 2 in json probably wouldn't be super useful. but if you want to fix it you'll have to make custom types for them i suppose

Reply:
_nods_ yeah, likely similar to how I made the custom type for serialising the reminder JSON itself.

I want to look into them a bit more to see what the specific data included is, as knowing that a reminder is repeating is pretty useful for me when exporting JSON (as my main  use case is likely going to be to use that JSON to load my reminders into another todolist app). I'm not 100% what the alarm actually relates to yet, it might just be similar to the `dueDate`, in which case it would be somewhat irrelevant.